### PR TITLE
Support for crowdloan refund

### PR DESCRIPTION
**Pull request by jacogr**
https://github.com/paritytech/polkadot/pull/2770

Does need type changes as well... fields removed....

### CODE REVIEW

**Comment by shawntabrizi** [comment]
@jacogr dunno if i will push the field removal now or later. Dont want to have to migrate on Rococo...

**Comment by jacogr** [comment]
Prefer with the field in :)

File: packages/page-parachains/src/Crowdloan/Refund.tsx [hunk 1]
```diff
@@ -125,29 +125,43 @@ function Withdraw ({ allAccounts, className, myAccounts, paraId }: Props): React
...
             </Modal.Columns>
-            <Modal.Columns hint={t<string>('The type of withdrawal, all accounts or only own accounts (if available)')}>
-              <Dropdown
-                label={t<string>('withdrawal type')}
-                onChange={setWithdrawType}
-                options={typeOptions}
-                value={withdrawType}
-              />
-            </Modal.Columns>
+            {api.tx.crowdloan.refund && (
```
Comment (jacogr) [inline packages/page-parachains/src/Crowdloan/Refund.tsx:128]:
```suggestion
            {!api.tx.crowdloan.refund && (
```

File: packages/page-parachains/src/Crowdloan/Fund.tsx [hunk 2]
```diff
@@ -43,14 +43,19 @@ function extractContributors (allAccounts: string[], keys: StorageKey[]): Contri
...
-  const trigger = useEventTrigger([api.events.crowdloan?.Contributed, api.events.crowdloan?.Withdrew], useCallback(
-    ({ event: { data: [, fundIndex] } }: EventRecord) =>
-      (fundIndex as ParaId).eq(paraId),
+  const trigger = useEventTrigger([api.events.crowdloan?.Contributed, api.events.crowdloan?.Withdrew, api.events.crowdloan?.AllRefunded, api.events.crowdloan?.PartiallyRefunded], useCallback(
+    ({ event: { data } }: EventRecord) =>
+      ((data.length === 1
+        // AllRefunded, PartiallyRefunded [ParaId]
+        ? data[0]
+        // Contributed, Withdrew [AccountId, ParaId, Balance]
+        : data[1]
```
Comment (jacogr) [inline packages/page-parachains/src/Crowdloan/Fund.tsx:57]:
```suggestion
        ? data[0] // AllRefunded, PartiallyRefunded [ParaId]
        : data[1] // Contributed, Withdrew [AccountId, ParaId, Balance]
```

**Pull request merged**